program/src/instruction.rs

```diff
@@ -1,184 +1,373 @@
+    )]
     StakeTokens(u64),
 
     /// Deactivate staked tokens for the validator.
-    ///
-    /// Only one deactivation may be in-flight at once, so if this is called
-    /// with an active deactivation, it will succeed, but reset the amount and
-    /// timestamp.
-    ///
-    /// 0. `[w]` Validator stake account
-    /// 1. `[s]` Authority on validator stake account
-    ///
-    /// Instruction data: amount of tokens to deactivate, as a little-endian u64
+    #[account(
+        0,
+        writable,
+        name = "stake",
+        desc = "Validator stake account"
+    )]
+    #[account(
+        1,
+        signer,
+        name = "stake_authority",
+        desc = "Authority on validator stake account"
+    )]
     DeactivateStake(u64),
 
     /// Move tokens from deactivating to inactive.
-    ///
-    /// Reduces the total voting power for the stake account and the total staked
-    /// amount on the system.
-    ///
-    /// NOTE: This instruction is permissionless, so anybody can finish
-    /// deactivating someone's tokens, preparing them to be withdrawn.
-    ///
-    /// 0. `[w]` Stake config account
-    /// 1. `[w]` Validator stake account
+    #[account(
+        0,
+        writable,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        writable,
+        name = "stake",
+        desc = "Validator stake account"
+    )]
     InactivateStake,
 
-    /// Withdraw inactive staked tokens from the vault
-    ///
-    /// After a deactivation has gone through the cooldown period and been
-    /// "inactivated", the authority may move the tokens out of the vault.
-    ///
-    /// 0. `[w]` Config account
-    /// 1. `[w]` Stake account
-    /// 2. `[w]` Vault token account
-    /// 3. `[w]` Destination token account
-    /// 4. `[s]` Stake authority
-    /// 5. `[]` Vault authority, PDA with seeds `['token-owner', stake_config]`
-    /// 6. `[]` SPL Token program
-    /// 7.. Extra required accounts for transfer hook
-    ///
-    /// Instruction data: amount of tokens to move
+    /// Withdraw inactive staked tokens from the vault.
+    #[account(
+        0,
+        writable,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        writable,
+        name = "stake",
+        desc = "Stake account"
+    )]
+    #[account(
+        2,
+        signer,
+        name = "stake_authority",
+        desc = "Stake authority"
+    )]
+    #[account(
+        3,
+        name = "vault_authority",
+        desc = "Vault authority"
+    )]
+    #[account(
+        4,
+        writable,
+        name = "vault_token",
+        desc = "Vault token account"
+    )]
+    #[account(
+        5,
+        writable,
+        name = "destination_token",
+        desc = "Destination token account"
+    )]
+    #[account(
+        6,
+        name = "spl_token_program",
+        desc = "SPL Token program"
+    )]
```

Comment:
This one seems more generalized than the previous instruction which requires "Token-2022" specifically.

Reply:
Replied on the previous comment about this, but I definitely need to be consistent once we agree on an approach.